openapi-cli/src/main/java/org/ballerinalang/openapi/CodeGenerator.java

```diff
@@ -182,10 +181,15 @@ public void generateService(String executionPath, String definitionPath,
             throw new BallerinaOpenApiException("Couldn't read the definition from file: " + definitionPath);
         }
 
-        if (serviceName != null) {
-            api.getInfo().setTitle(serviceName);
-        } else if (api.getInfo() == null || StringUtils.isEmpty(api.getInfo().getTitle())) {
-            api.getInfo().setTitle(GeneratorConstants.UNTITLED_SERVICE);
+        if (api.getInfo() == null) {
+            throw new BallerinaOpenApiException("InFo section couldn't be null in the definition from file: " +
+                    definitionPath);
+        } else {
+            if (api.getInfo().getTitle().isBlank() && (serviceName == null || serviceName.isBlank())) {
+                api.getInfo().setTitle(GeneratorConstants.UNTITLED_SERVICE);
+            } else {
+                api.getInfo().setTitle(serviceName);
+            }
```

Comment:
This logic can be moved out of the `else` block, since the `if` block does an early exit.